kolena/_experimental/object_detection/utils.py

```diff
@@ -182,25 +182,18 @@ def compute_confusion_matrix_plot(
 def _compute_sklearn_arrays_by_class(
     all_matches: List[MulticlassInferenceMatches],
 ) -> Tuple[np.ndarray, np.ndarray, Dict[str, np.ndarray], Dict[str, np.ndarray]]:
-    y_true: List[int] = []
-    y_score: List[float] = []
+    y_true, y_score = _compute_sklearn_arrays(all_matches)
     y_true_by_label: defaultdict[str, List[int]] = defaultdict(lambda: [])
     y_score_by_label: defaultdict[str, List[float]] = defaultdict(lambda: [])
     for image_bbox_matches in all_matches:
         for _, bbox_inf in image_bbox_matches.matched:  # TP (if above threshold)
-            y_true.append(1)
-            y_score.append(bbox_inf.score)
             y_true_by_label[bbox_inf.label].append(1)
             y_score_by_label[bbox_inf.label].append(bbox_inf.score)
         for gt_or_pair in image_bbox_matches.unmatched_gt:  # FN
             gt_label: str = gt_or_pair[0].label if type(gt_or_pair) is tuple else gt_or_pair.label
-            y_true.append(1)
-            y_score.append(-1)
             y_true_by_label[gt_label].append(1)
             y_score_by_label[gt_label].append(-1)
         for bbox_inf in image_bbox_matches.unmatched_inf:  # FP (if above threshold)
-            y_true.append(0)
-            y_score.append(bbox_inf.score)
             y_true_by_label[bbox_inf.label].append(0)
             y_score_by_label[bbox_inf.label].append(bbox_inf.score)
```

Comment:
Structurally this still looks very similar to `_compute_sklearn_arrays` — how can we guarantee that `y_true` and `y_true_by_label` are equivalent when one is created upstream and the other is created here? Logic can be further refactored to share implementation.

Reply:
Good call!